File: shared/src/commonMain/kotlin/me/timeto/shared/extensions/Int.kt
package me.timeto.shared.extensions

fun Int.limitMin(value: Int): Int =
    if (this < value) value else this

fun Int.limitMax(value: Int): Int =
    if (this > value) value else this

fun Int.limitMinMax(min: Int, max: Int): Int =
    this.limitMin(min).limitMax(max)

File: apple/Common/Int.swift
extension Int {

    func limitMin(_ value: Int) -> Int {
        self < value ? value : self
    }

    func limitMax(_ value: Int) -> Int {
        self > value ? value : self
    }

    func limitMinMax(
        _ min: Int,
        _ max: Int
    ) -> Int {
        limitMin(min).limitMax(max)
    }
}
